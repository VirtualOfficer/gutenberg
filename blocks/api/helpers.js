/**
 * External dependencies
 */
import { every, keys, isEqual } from 'lodash';

/**
 * Internal dependencies
 */
import { getDefaultBlockName } from './registration';
import { createBlock } from './factory';

/**
 * Determines whether the block is a default block
 * and its attributes are equal to the default attributes
 * which means the block is untouched.
 *
 * @param  {WPBlock} block Block Object
 *
 * @return {boolean}       Whether the block is an untouched default block
 */
export function isUntouchedDefaultBlock( block ) {
	const newDefaultBlock = createBlock( getDefaultBlockName() );
	const attributeKeys = {
		...keys( newDefaultBlock.attributes ),
		...keys( block.attributes ),
	};

	return (
		block.name === newDefaultBlock.name &&
		every( attributeKeys, ( key ) => isEqual( newDefaultBlock.attributes[ key ], block.attributes[ key ] ) )
	);
}
