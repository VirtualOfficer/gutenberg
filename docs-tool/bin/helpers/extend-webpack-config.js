const path = require( 'path' );

module.exports = function( webpackConfig, usersCwd ) {
	// Adding "glutenberg" alias
	webpackConfig.resolve.alias.glutenberg = path.resolve( __dirname, '../../src/config/' );

	// Loading the config folder
	webpackConfig.resolve.alias.config = path.resolve( usersCwd, process.argv[ 2 ] );
	webpackConfig.resolve.modules = webpackConfig.resolve.modules.concat( [ webpackConfig.resolve.alias.config ] );

	// Using the user's node_modules
	const usersNodeModules = path.resolve( usersCwd, 'node_modules' );
	webpackConfig.resolve.modules = webpackConfig.resolve.modules.concat( [ usersNodeModules ] );

	// Deleting CRA scoping
	webpackConfig.resolve.plugins = [];
	webpackConfig.module.rules.forEach( ( rule ) => {
		if ( rule.include ) {
			rule.include = [ rule.include, webpackConfig.resolve.alias.config ];
		}
	} );

	// Adding the markdown loader and exclude if from the file loader
	webpackConfig.module.rules.forEach( rule => {
		if ( rule.loader === require.resolve('file-loader') ) {
			rule.exclude.push( /\.md/ );
		}
	} );
	webpackConfig.module.rules.push( {
		test: /\.md/,
		use: require.resolve( 'raw-loader' ),
	} );
}
