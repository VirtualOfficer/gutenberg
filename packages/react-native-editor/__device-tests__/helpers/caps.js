exports.ios = {
	browserName: '',
	platformName: 'iOS',
	platformVersion: '13.0',
	deviceName: 'iPhone 11 Simulator',
	os: 'iOS',
	deviceOrientation: 'portrait',
	automationName: 'XCUITest',
	appiumVersion: '1.15.0', // SauceLabs requires appiumVersion to be specified.
	app: undefined, // will be set later, locally this is relative to root of project
};

exports.android8 = {
	browserName: '',
	platformName: 'Android',
	platformVersion: '9.0',
	deviceName: 'Google Pixel 3 XL GoogleAPI Emulator',
	automationName: 'UiAutomator2',
	os: 'Android',
	appPackage: 'com.gutenberg',
	appActivity: 'com.gutenberg.MainActivity',
	deviceOrientation: 'portrait',
	appiumVersion: '1.15.0',
	app: undefined,
};
