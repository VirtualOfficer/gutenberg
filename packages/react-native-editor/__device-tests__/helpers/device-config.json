{
	"ios": {
		"local": {
			"deviceName": "iPhone 15",
			"deviceTabletName": "iPad (10th generation)",
			"platformVersion": "17.0"
		},
		"github": {
			"deviceName": "iPhone 14",
			"deviceTabletName": "iPad (10th generation)",
			"platformVersion": "16.2"
		},
		"pixelRatio": {
			"iPhone": 3,
			"iPad": 2
		}
	},
	"android": {
		"local": {
			"deviceName": "Pixel_3_XL_API_30",
			"platformVersion": "11.0"
		},
		"saucelabs": {
			"deviceName": "Google Pixel 3 XL GoogleAPI Emulator"
		}
	}
}
