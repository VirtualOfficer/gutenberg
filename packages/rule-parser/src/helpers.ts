/**
 * Internal dependencies
 */
import type { Value, RawRule, Rule, Rules, StrictRules } from './types';
import { combinators } from './constants';

function isPrimitive( value: unknown ): value is Value {
	return (
		typeof value === 'string' ||
		typeof value === 'number' ||
		typeof value === 'boolean'
	);
}

function isPrimitiveArray( value: unknown ): value is Array< Value > {
	if ( ! Array.isArray( value ) ) {
		return false;
	}
	const type = typeof value[ 0 ];
	return value.every( ( item ) => typeof item === type );
}

export function valueType( value: Value | Value[] ): string {
	return Array.isArray( value ) ? typeof value[ 0 ] : typeof value;
}

export function isRawRule( rule: unknown ): rule is RawRule {
	if ( ! Array.isArray( rule ) ) {
		return false;
	}

	if ( rule.length !== 3 ) {
		return false;
	}

	const [ source, operator, target ] = rule;

	if ( typeof source !== 'string' ) {
		return false;
	}

	if ( typeof operator !== 'string' ) {
		return false;
	}

	if ( ! isPrimitive( target ) && ! isPrimitiveArray( target ) ) {
		return false;
	}

	return true;
}

export function isRule( rule: unknown ): rule is Rule {
	if ( ! Array.isArray( rule ) ) {
		return false;
	}

	if ( rule.length !== 3 ) {
		return false;
	}

	const [ source, operator, target ] = rule;

	if ( ! isPrimitive( source ) && ! isPrimitiveArray( source ) ) {
		return false;
	}

	if ( typeof operator !== 'string' ) {
		return false;
	}

	if ( ! isPrimitive( target ) && ! isPrimitiveArray( target ) ) {
		return false;
	}

	return true;
}

export function stringifyRule( rule: Rule ): string {
	const [ source, operator, target ] = rule;
	const sourceString = Array.isArray( source )
		? `[ ${ source.join( ', ' ) } ]`
		: source;
	const targetString = Array.isArray( target )
		? `[ ${ target.join( ', ' ) } ]`
		: target;

	return `[ ${ sourceString } ${ operator } ${ targetString } ]`;
}

export function isStructuredRule< T extends Rule | boolean >(
	rules: Rules< T >
): rules is StrictRules< T > {
	return (
		Array.isArray( rules ) &&
		typeof rules[ 0 ] === 'string' &&
		combinators.includes( rules[ 0 ] )
	);
}
