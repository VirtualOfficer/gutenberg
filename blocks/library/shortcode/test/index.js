/**
 * Internal dependencies
 */
import { registerShortcodeBlock } from '../';
import { blockEditRender } from 'blocks/test/helpers';

describe( 'core/shortcode', () => {
	beforeAll( () => {
		registerShortcodeBlock();
	} );

	test( 'block edit matches snapshot', () => {
		const wrapper = blockEditRender( 'core/shortcode' );

		expect( wrapper ).toMatchSnapshot();
	} );
} );
