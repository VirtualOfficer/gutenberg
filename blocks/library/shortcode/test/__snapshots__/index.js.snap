// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core/shortcode block edit matches snapshot 1`] = `
<div
  class="wp-block"
>
  <div
    class="wp-block-embed is-loading"
  >
    <span
      class="spinner is-active"
    />
    <p>
      Loading...
    </p>
  </div>
</div>
`;
